src/lib/factory.ts

```diff
@@ -1,12 +1,18 @@
 import { ClassType, ConfigStorage } from './types';
-import { ENV_CONFIG_NAME_SYMBOL } from './symbols';
+import { ENV_CONFIG_NAME_SYMBOL, UNSCOPED_CONFIG_SYMBOL } from './symbols';
 import { plainToClass } from '../transformer';
 
 export class ConfigFactory {
   public createConfig(
     configStorage: ConfigStorage,
     ConfigClass: ClassType,
   ): typeof ConfigClass.prototype {
+    if (ConfigClass[UNSCOPED_CONFIG_SYMBOL]) {
+      return plainToClass(ConfigClass, configStorage[UNSCOPED_CONFIG_SYMBOL], {
+        excludeExtraneousValues: true,
```

Comment:
there will be different behaviour with `@UnscopedConfig` and usuall `@Config`. So after migration could be implicit consequences

Reply:
Yep, but without excludeExtraneousValues all envs will be automatically exposed to class instance which is more terrible than the inconsistent behavior. Moreover, I do not understand why this option is not enabled in the `@Config` mode. And, I hope, `@UnscopedConfig` is more for migration cases than for everyday use, because implementation is one big workaround.

I think we should describe `excludeExtraneousValues` behavior in readme of `@UnscopedConfig` and enable it for `@Config` in next major version.

P.S. But in general, you gave me a strange idea how to reduce workaroundness of this solution, I'll try to implement it.